File: modules/AcroStack.Chat/BlockedUserDto.cs
using System;
using Volo.Abp.Application.Dtos;

namespace AcroStack.Chat;

/// <summary>A user blocked by the current user in the chat module.</summary>
public class BlockedUserDto : EntityDto<Guid>
{
    /// <summary>IdentityUser.Id of the blocked user.</summary>
    public Guid BlockedUserId { get; set; }

    /// <summary>User name of the blocked user.</summary>
    public string BlockedUserName { get; set; } = string.Empty;

    /// <summary>When the block was created.</summary>
    public DateTime CreationTime { get; set; }
}

File: react/src/api/models/acroStack/chat/BlockedUserDto.ts
/* oxlint-disable */

/**
 * @description A user blocked by the current user in the chat module.
 * @type object
 */
export type AcroStackChatBlockedUserDto = {
  /**
   * @description
   * Format: `uuid`
   * @type string | undefined
   */
  id?: string;
  /**
   * @description IdentityUser.Id of the blocked user.
   *
   * Format: `uuid`
   * @type string | undefined
   */
  blockedUserId?: string;
  /**
   * @description User name of the blocked user.
   * @type string | undefined
   */
  blockedUserName?: string | null;
  /**
   * @description When the block was created.
   *
   * Format: `date-time`
   * @type string | undefined
   */
  creationTime?: string;
};
